pkg/providers/vsphere/vsphere.go

```diff
@@ -695,7 +695,7 @@ func (p *vsphereProvider) GenerateCAPISpecForCreate(ctx context.Context, _ *type
 }
 
 func (p *vsphereProvider) GenerateStorageClass() []byte {
-	if p.datacenterConfig.Spec.DisableCSI {
+	if !p.csiEnabled {
```

Comment:
This is not ideal, we should probably pass the cluster spec to this method
But that would require wider changes, affecting all providers. At least this removed the dependency and extra copies of the vsphere datacenter